packages/core/src/Actor.ts

```diff
@@ -49,57 +56,327 @@ export function createNullActor(id: string): ActorRef<any> {
-    tempActor.getSnapshot = () => initialState;
-  }
+): { start: (...args: any[]) => void; actorRef: ActorRef<any, any> } {
+  const { name } = options;
+  let canceled = false;
+  let emitted = data;
+  let queued: any[] | null = [];
+  let parent: any;
+
+  let subscribers = new Set<DisposableObserver>();
+  let receivers = new Set<{ send: (ev: any) => void }>();
+  let boxedEntity:
+    | {
+        stop?: () => void;
+        send?: (ev: any) => void;
+        getSnapshot?: () => any;
+        subscribe?: (observer: DisposableObserver) => void;
+      }
+    | undefined = initialEntity;
+
+  const actorRef = initialEntity || {
+    id: name,
+    send: (ev) => {
+      if (queued) {
+        queued.push(ev);
+        return;
+      }
+      if (boxedEntity && 'send' in boxedEntity) {
+        boxedEntity.send!(
+          wrappedMachines.has(actorRef)
+            ? wrapWithOrigin(parent, toSCXMLEvent(ev))
+            : ev
+        );
+        return;
+      }
+      receivers.forEach((receiver) => receiver.send(ev));
+    },
+    subscribe: (next, handleError?, complete?) => {
+      const observer: DisposableObserver = toObserver(
+        next,
+        handleError,
+        complete
+      );
+      subscribers.add(observer);
+
+      boxedEntity?.subscribe?.(observer);
+
+      return {
+        unsubscribe: () => {
+          subscribers.delete(observer);
+          observer.unsubscribe?.();
+        }
+      };
+    },
+    stop: () => {
+      canceled = true;
+      boxedEntity?.stop?.();
+    },
+    toJSON() {
+      return { id: name };
+    },
+    getSnapshot: () => {
+      if (boxedEntity && 'getSnapshot' in boxedEntity) {
+        return boxedEntity.getSnapshot!();
+      }
+      return emitted;
+    },
+    [symbolObservable]: function () {
+      return this;
+    }
+  };
+
+  return {
+    actorRef,
+    start({ parent: _parent, entity }: any) {
+      parent = _parent;
+      // defensive measure, ideally `_start` should never be called in such a state
+      if (canceled) {
+        return;
+      }
+      parent.children.set(name, actorRef);
 
-  return tempActor;
+      let entityOrStop: any;
+
+      if (initialEntity) {
+        // skip
+      } else if (isMachine(entity) || isBehavior(entity)) {
+        entityOrStop = entity;
+      } else {
+        try {
+          const callbackSend = (e: typeof emitted) => {
+            emitted = e;
+            subscribers.forEach((observer) => observer.next(e));
+            if (canceled) {
+              return;
+            }
+            parent.send(toSCXMLEvent(e, { origin: name }));
+          };
+          const callbackOnReceive = (newListener) => {
+            receivers.add({
+              send: newListener
+            });
+          };
+
+          entityOrStop = entity(callbackSend, callbackOnReceive);
+        } catch (err) {
+          parent.send(error(name, err));
+          return;
+        }
+      }
+
+      const startInterpreter = (service) => {
```

Comment:
the problem with machines is that we currently wrap them with that "deferred actor" object - this means that what we hand to the user is no longer an interpreter, it's that deferred actor object. At the type-level, nothing changes because those things (`state.children`, return type of `spawn`, etc) were already typed as ActorRefs - but at the runtime level we effectively "remove" things (properties, methods) from here.